app/src/main/java/de/westnordost/streetcomplete/quests/shop_type/SpecifyShopType.kt

```diff
@@ -0,0 +1,64 @@
+package de.westnordost.streetcomplete.quests.shop_type
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.LAST_CHECK_DATE_KEYS
+import de.westnordost.streetcomplete.data.meta.SURVEY_MARK_KEY
+import de.westnordost.streetcomplete.data.meta.toCheckDateString
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.osmquest.OsmFilterQuestType
+import de.westnordost.streetcomplete.ktx.containsAny
+import java.util.*
+
+class SpecifyShopType : OsmFilterQuestType<ShopTypeAnswer>() {
+
+    override val elementFilter = """
+        nodes, ways, relations with (
+             shop = yes
+             and !man_made
+             and !historic
+             and !military
+             and !power
+             and !tourism
+             and !attraction
+             and !amenity
```

Comment:
```suggestion
             and !amenity
             and !craft
             and !tourism
```

For example if someone tagged `craft=shoemaker + shop=yes` to denote that this is a shoemaker you can just walk in as opposed to a workshop of a shoemaker that is not also a shop.
Or for example, a museum and somebody put `shop=yes` on it too, to denote that there is also a museum (gift) shop.

(you mentioned you prefer overkill over whack-a-mole)